### PR TITLE
Announce bazel 0.16 release

### PR DESCRIPTION
**Pull request by c-parsons**

### CODE REVIEW

File: _posts/2018-07-31-bazel-0.16.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+layout: posts
+title: "Bazel 0.16"
+authors:
+  - cparsons
```
Comment (laurentlb) [inline _posts/2018-07-31-bazel-0.16.md]: I think you need to be in this list too:
https://github.com/bazelbuild/bazel-blog/blob/master/_config.yml

Comment (c-parsons) [reply _posts/2018-07-31-bazel-0.16.md]: Requested in https://github.com/bazelbuild/bazel-blog/pull/109

File: _posts/2018-07-31-bazel-0.16.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+The Bazel team is happy to announce a new version of Bazel, [Bazel 0.16](https://github.com/bazelbuild/bazel/releases/tag/0.16.0)!
+
+We’ve updated our design process. All design documents are now published on a separate GitHub repository. They are explicitly approved via GitHub pull requests, so that you can easily track new proposals and status changes.
```
Comment (jin) [inline _posts/2018-07-31-bazel-0.16.md]: add the link to the design process and repository?

File: _posts/2018-07-31-bazel-0.16.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Here are the main changes of Bazel 0.16:
+
+
+## General changes
+
+*    Bazel now allows `@` in package names. For example, `node_modules/@namespace_foo/lib_bar` (which is a common pattern for NPM) is now an allowed package name.

+*    The `--distdir` option is no longer experimental. This option allows to specify additional directories to look for files before trying to fetch them from the network. Files from any of the distdirs are only used if a checksum for the file is specified, and both the filename and the checksum match.
```
Comment (jin) [inline _posts/2018-07-31-bazel-0.16.md]: please link the flag to https://docs.bazel.build/versions/master/command-line-reference.html#flag--distdir

File: _posts/2018-07-31-bazel-0.16.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+*    `--noexpand_configs_in_place` is now deprecated, after a transition period to inplace-expansion, see https://blog.bazel.build/2018/01/19/config-parsing-order.html for details
```
Comment (jin) [inline _posts/2018-07-31-bazel-0.16.md]: please linkify

**Comment by laurentlb** [comment]
To avoid losing links, use a script to convert the Google Doc to markdown (I know we have one inside Google).

**Comment by c-parsons** [comment]
Good catch, thanks! Done.

**Comment by jin** [comment]
@cushon @buchgr can we make a note about the changes to the embedded JDK? it's invisible to the users, but considering that it breaks users with custom --host_javabase flags, we should explain what's going on behind the scenes.

**Comment by cushon** [comment]
Maybe something like:

> The embedded JDK has been upgraded to JDK 9, and is now used as both the default [startup `--host_javabase`](https://docs.bazel.build/versions/master/user-manual.html#startup_flag--host_javabase) and [build `--host_javabase`](https://docs.bazel.build/versions/master/user-manual.html#flag--host_javabase).

**Comment by cushon** [comment]
@buchgr to the extent that JDK 8 is still a supported host_javabase, the toolchain selection is supposed to be automated:

https://github.com/bazelbuild/bazel/blob/f621bbe84e8c2bff47d8d060f28be2a8b5bae834/src/main/java/com/google/devtools/build/lib/bazel/rules/GenericRules.java#L50

**Comment by jin** [comment]
Since @c-parsons is OOO, let's merge the changelog in and publish the post first. We'll add the note about the JDK changes in another PR.

**Pull request merged**